pcdsdevices/epics_motor.py

```diff
@@ -197,6 +224,26 @@ def _get_epics_limits(self) -> tuple[float, float]:
             return (0, 0)
         return limits
 
+    @property
+    def msta(self):
+        """
+        Returns the msta fields as a dictionary.
+        """
+        # the MSTA field is a float for some reason...
+        val = int(self.msta_raw.get())
+        d = dict()
+        for bit in MstaEnum:
+            d[bit.name] = (val >> bit.value) & 0x1
+        return d
```

Comment:
```python
d = {bit.name: (val >> bit.value) & 0x1 for bit in MstaEnum}  
```
A fun pythonic dict-comprehension could make this a one-liner.

A more meaningful comment: I wonder if there's a clearer way to do these bit shifts.  It took my python mind a while to understand the simple bit-extraction you're doing here.  

Maybe `enum.IntFlag`?
```python
from enum import IntFlag, auto

class MstaEnum(IntFlag):
    """
    Enum for the EPICS motor record .MSTA field bits.
    """
    # Note: the motor record docs start bit numbering at 1, but the MSTA bits
    # start at 0.
    direction = auto()       # last raw move direction (0: negative, 1: positive)
    done = auto()            # motion is complete
    plus_ls = auto()         # plus limit switch is hit
    home_ls = auto()         # state of the home limit switch
    # Bit 4 is un-used
    closed_loop = auto()
    slip_stall = auto()      # slip stall is detected
    home = auto()            # at the home position
    enc_present = auto()     # encoder is present. Called "present" in the docs.
    problem = auto()         # driver stopped polling, or there's a hardware problem
    moving = auto()         # the motor has a non-zero velocity (it's moving)
    gain_support = auto()   # the motor supports closed loop control
    comm_error = auto()     # controller communication error
    minus_ls = auto()       # minus limit switch is hit
    homed = auto()          # the motor has been homed


print(MstaEnum(0))  # >> MstaEnum.0
print(MstaEnum(129))  # >> MstaEnum.enc_present|direction

print([flag.name for flag in MstaEnum if flag in MstaEnum(129)])  # >> ['direction', 'enc_present]
```

Reply:
That's a good one-liner. I forgot about dictionary comprehensions. Sometimes my python ends up a bit C-ish. :)

Does `auto()` work here? I'm not familiar with it but from the docs it sounds like it just generates sequential numbers. The raw MSTA value that we're getting from the IOC is based on the mapping in the motor record, so the values matter, and I think we'd get off by one after the missing 4th bit.

Reply:
I think to use auto (and an int flag in general) you'd have to include a placeholder bit

Reply:
Now that I see the IMS and Newport IOCs have members that are larger than one bit, I think `auto()` will not work, or at least not easily or cleanly. I think I'd rather stick with an explicit mapping rather than implicitly making it with `auto()`.

Reply:
I think `IntFlag` still works if you define the bits manually, but I agree `auto()` won't quite work here.
```python
class MstaEnum(IntFlag):
    """
    Enum for the EPICS motor record .MSTA field bits.
    """
    # Note: the motor record docs start bit numbering at 1, but the MSTA bits
    # start at 0.
    direction = 1       # last raw move direction (0: negative, 1: positive)
    done = 2            # motion is complete
    plus_ls = 4        # plus limit switch is hit
    home_ls = 8         #
    ...
```